src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -851,6 +852,13 @@ CreateHashDistributedTableShards(Oid relationId, int shardCount,
 
 	if (colocatedTableId != InvalidOid)
 	{
+		/*
+		 * If there is a concurrent shard move on this
+		 * colocation group fail this backend. But, allow
+		 * concurrent colocated table creation.
```

Comment:
is concurrent table creation always safe (e.g. also in case of foreign keys?)

Reply:
Hmm, could work.

Should we then change the behaviour to `wait` instead of throwing an error? At least, for this part of the code.

Reply:
I'm mainly wondering whether it is always safe to run 2 create_distributed_table calls in the same co-location group concurrently, and not sure about the answer.

Reply:
probably not safe, I'm assuming you are considering something like the below, which would most likely fail:

create table t1 (a int, ..)
create_dist_table(t1, ..);

create table t2 (a int, ..)
create table t3 (a int, ..)
ALTER TABLE t2 ADD FOREIGN KEY references t3;
ALTER TABLE t3 ADD FOREIGN KEY references t2;




Session 1:
```
create_dist_table(t2, colocate_with:t1);
```

session 2:
```
create_dist_table(t3, colocate_with:t1);
```